pvlib/pvsystem.py

```diff
@@ -66,6 +69,57 @@ def _combine_localized_attributes(pvsystem=None, location=None, **kwargs):
     return new_kwargs
 
 
+def return_singleton_value(func):
```

Comment:
We should put these decorators in tools.py (which is not part of the public api) or we should make them private here.

Reply:
Similar in spirit, but I think sufficiently different in practice that a custom solution is warranted. `squeeze([x])` will return a singleton array and `ndarray.item` will only operate on a singleton array (if no index is given). We would still need to add our own code to handle returning an array if the result was not a singleton.

> We should put these decorators in tools.py (which is not part of the public api) or we should make them private here.

I think private for now, not sure they are really broadly applicable at this point, unless @cwhanse wants to use them in `ModelChain`. By applying them in `PVSystem` I think we have avoided needing to do that though.

Reply:
Can't take a close look at this PR right now but I will sometime soon.  Just hopping in to say that I think the way the term "singleton" is used here is not wholly consistent with the meaning I'm familiar with in CS: https://en.wikipedia.org/wiki/Singleton_pattern

Reply:
@kanderso-nrel That's a good point, I'm not sure what I was thinking of when I named it, but the design pattern is by far the most common usage for the term.